### PR TITLE
rework walk to proxy

### PR DESCRIPTION
**Pull request by Raycoms**
This allows us to use our proxy system also for not worker entities inside our colonies.
It also improves the readability a bit.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractWalkToProxy.java [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+    /**
+     * Leads the worker to a certain position due to proxies.
+     *
+     * @param target the target position.
+     * @param range  the range.
+     * @param onMove worker on move or not?
+     * @return true if arrived.
+     */
+    public boolean walkToBlock(@NotNull final BlockPos target, final int range, final boolean onMove)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractWalkToProxy.java]: Always had, but this is not the job of this pr.

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/EntityCitizenWalkToProxy.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+    /**
+     * Returns a proxy point to the goal for the miner especially.
+     *
+     * @param target         the target.
+     * @param distanceToPath the total distance.
+     * @return a proxy or, if not applicable null.
+     */
+    @NotNull
+    private BlockPos getMinerProxy(final BlockPos target, final double distanceToPath, @NotNull final BuildingMiner building)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/pathfinding/EntityCitizenWalkToProxy.java:77]: MAJOR The Cyclomatic Complexity of this method "getMinerProxy" is 21 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
MAJOR Refactor this method to reduce its Cognitive Complexity from 42 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 62 issues
* ![MAJOR][MAJOR] 28 major
* ![MINOR][MINOR] 31 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityCitizen.java](https://github.com/Minecolonies/minecolonies/blob/fa62a851d6909d089cad1618a1e826a569dd794a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java): This file has 1.292 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/fa62a851d6909d089cad1618a1e826a569dd794a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 57 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/fa62a851d6909d089cad1618a1e826a569dd794a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): class "EntityCitizen" has 94 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L350](https://github.com/Minecolonies/minecolonies/blob/fa62a851d6909d089cad1618a1e826a569dd794a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L350): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L352](https://github.com/Minecolonies/minecolonies/blob/fa62a851d6909d089cad1618a1e826a569dd794a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L352): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L353](https://github.com/Minecolonies/minecolonies/blob/fa62a851d6909d089cad1618a1e826a569dd794a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L353): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L354](https://github.com/Minecolonies/minecolonies/blob/fa62a851d6909d089cad1618a1e826a569dd794a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L354): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L356](https://github.com/Minecolonies/minecolonies/blob/fa62a851d6909d089cad1618a1e826a569dd794a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L356): Assign this magic number 3.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L356](https://github.com/Minecolonies/minecolonies/blob/fa62a851d6909d089cad1618a1e826a569dd794a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L356): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L357](https://github.com/Minecolonies/minecolonies/blob/fa62a851d6909d089cad1618a1e826a569dd794a/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L357): Assign this magic number 5.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by xavierh — APPROVED** [review]

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 62 issues
* ![MAJOR][MAJOR] 28 major
* ![MINOR][MINOR] 31 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityCitizen.java](https://github.com/Minecolonies/minecolonies/blob/1ef344b6b7e379364561c89c512c0d704358affb/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java): This file has 1.292 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/1ef344b6b7e379364561c89c512c0d704358affb/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 57 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/1ef344b6b7e379364561c89c512c0d704358affb/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): class "EntityCitizen" has 94 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L350](https://github.com/Minecolonies/minecolonies/blob/1ef344b6b7e379364561c89c512c0d704358affb/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L350): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L352](https://github.com/Minecolonies/minecolonies/blob/1ef344b6b7e379364561c89c512c0d704358affb/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L352): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L353](https://github.com/Minecolonies/minecolonies/blob/1ef344b6b7e379364561c89c512c0d704358affb/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L353): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L354](https://github.com/Minecolonies/minecolonies/blob/1ef344b6b7e379364561c89c512c0d704358affb/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L354): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L356](https://github.com/Minecolonies/minecolonies/blob/1ef344b6b7e379364561c89c512c0d704358affb/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L356): Assign this magic number 3.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L356](https://github.com/Minecolonies/minecolonies/blob/1ef344b6b7e379364561c89c512c0d704358affb/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L356): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L357](https://github.com/Minecolonies/minecolonies/blob/1ef344b6b7e379364561c89c512c0d704358affb/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L357): Assign this magic number 5.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractWalkToProxy.java [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Abstract class for the walkToProxy.
+ */
+public abstract class AbstractWalkToProxy
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractWalkToProxy.java]: At least define an Interface for this class in the API.
If it is possible move it completly to the API package as well next to the creation of the Interface.

**Review by marchermans — CHANGES_REQUESTED** [review]

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractWalkToProxy.java (hunk 3 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractWalkToProxy.java]: Sure I moved it to the APi Package, the abstractWalkToProxy has no dependencies on minecolonies.
Only on:
import com.minecolonies.api.util.BlockPosUtil;
import com.minecolonies.api.util.EntityUtils;
import net.minecraft.entity.EntityLiving;
import net.minecraft.util.math.BlockPos;

**Comment by Raycoms** [comment]
@OrionDevelopment moved it

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 62 issues
* ![MAJOR][MAJOR] 28 major
* ![MINOR][MINOR] 31 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityCitizen.java](https://github.com/Minecolonies/minecolonies/blob/b3326ff5263598d2c015c485e9d0635a6ec5ce44/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java): This file has 1.292 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/b3326ff5263598d2c015c485e9d0635a6ec5ce44/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 57 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/b3326ff5263598d2c015c485e9d0635a6ec5ce44/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): class "EntityCitizen" has 94 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L350](https://github.com/Minecolonies/minecolonies/blob/b3326ff5263598d2c015c485e9d0635a6ec5ce44/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L350): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L352](https://github.com/Minecolonies/minecolonies/blob/b3326ff5263598d2c015c485e9d0635a6ec5ce44/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L352): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L353](https://github.com/Minecolonies/minecolonies/blob/b3326ff5263598d2c015c485e9d0635a6ec5ce44/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L353): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L354](https://github.com/Minecolonies/minecolonies/blob/b3326ff5263598d2c015c485e9d0635a6ec5ce44/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L354): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L356](https://github.com/Minecolonies/minecolonies/blob/b3326ff5263598d2c015c485e9d0635a6ec5ce44/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L356): Assign this magic number 3.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L356](https://github.com/Minecolonies/minecolonies/blob/b3326ff5263598d2c015c485e9d0635a6ec5ce44/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L356): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L357](https://github.com/Minecolonies/minecolonies/blob/b3326ff5263598d2c015c485e9d0635a6ec5ce44/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L357): Assign this magic number 5.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marchermans** [comment]
Did you also create an interface for it? Or just move it.

**Comment by Raycoms** [comment]
I just moved it, I understood or interface or moving it?

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 62 issues
* ![MAJOR][MAJOR] 28 major
* ![MINOR][MINOR] 31 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityCitizen.java](https://github.com/Minecolonies/minecolonies/blob/017b57405635c34eaddeb8797c0e0bcb6eb1776b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java): This file has 1.292 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/017b57405635c34eaddeb8797c0e0bcb6eb1776b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 57 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L69](https://github.com/Minecolonies/minecolonies/blob/017b57405635c34eaddeb8797c0e0bcb6eb1776b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L69): class "EntityCitizen" has 94 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L350](https://github.com/Minecolonies/minecolonies/blob/017b57405635c34eaddeb8797c0e0bcb6eb1776b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L350): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L352](https://github.com/Minecolonies/minecolonies/blob/017b57405635c34eaddeb8797c0e0bcb6eb1776b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L352): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L353](https://github.com/Minecolonies/minecolonies/blob/017b57405635c34eaddeb8797c0e0bcb6eb1776b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L353): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L354](https://github.com/Minecolonies/minecolonies/blob/017b57405635c34eaddeb8797c0e0bcb6eb1776b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L354): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L356](https://github.com/Minecolonies/minecolonies/blob/017b57405635c34eaddeb8797c0e0bcb6eb1776b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L356): Assign this magic number 3.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L356](https://github.com/Minecolonies/minecolonies/blob/017b57405635c34eaddeb8797c0e0bcb6eb1776b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L356): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L357](https://github.com/Minecolonies/minecolonies/blob/017b57405635c34eaddeb8797c0e0bcb6eb1776b/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L357): Assign this magic number 5.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]
This is okey.
Did you replace any direct usage of AbstractWalkToProxy with the new Interface, so that any other WalkToProxy classes (that might be added in the future, and cannot use the Abstract variant for what ever reason) work directly?

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 61 issues
* ![MAJOR][MAJOR] 28 major
* ![MINOR][MINOR] 30 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityCitizen.java](https://github.com/Minecolonies/minecolonies/blob/6d0c76bce4d3f55b00f8eec4742517ff810295f5/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java): This file has 1.293 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L70](https://github.com/Minecolonies/minecolonies/blob/6d0c76bce4d3f55b00f8eec4742517ff810295f5/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L70): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 57 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L70](https://github.com/Minecolonies/minecolonies/blob/6d0c76bce4d3f55b00f8eec4742517ff810295f5/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L70): class "EntityCitizen" has 94 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L351](https://github.com/Minecolonies/minecolonies/blob/6d0c76bce4d3f55b00f8eec4742517ff810295f5/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L351): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L353](https://github.com/Minecolonies/minecolonies/blob/6d0c76bce4d3f55b00f8eec4742517ff810295f5/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L353): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L354](https://github.com/Minecolonies/minecolonies/blob/6d0c76bce4d3f55b00f8eec4742517ff810295f5/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L354): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L355](https://github.com/Minecolonies/minecolonies/blob/6d0c76bce4d3f55b00f8eec4742517ff810295f5/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L355): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L357](https://github.com/Minecolonies/minecolonies/blob/6d0c76bce4d3f55b00f8eec4742517ff810295f5/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L357): Assign this magic number 3.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L357](https://github.com/Minecolonies/minecolonies/blob/6d0c76bce4d3f55b00f8eec4742517ff810295f5/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L357): Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)
1. ![MAJOR][MAJOR] [EntityCitizen.java#L358](https://github.com/Minecolonies/minecolonies/blob/6d0c76bce4d3f55b00f8eec4742517ff810295f5/src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java#L358): Assign this magic number 5.0F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]
Nice!

**Comment by Raycoms** [comment]
Codacy won't be happy with the complexity, but I actually decreased it, so I'll merge it.

**Pull request merged**